File: app/src/main/java/com/jesusc24/xroadsthroughthecastle/data/dao/ChatDAO.java
package com.jesusc24.xroadsthroughthecastle.data.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.jesusc24.xroadsthroughthecastle.data.model.Chat;

import java.util.List;

@Dao
public interface ChatDAO {

    @Insert
    long insert(Chat chat);

    @Update
    void update(Chat chat);

    @Delete
    void delete(Chat chat);

    @Query("delete from chat")
    void deleteAll();

    @Query("select * from chat order by nombre asc")
    List<Chat> select();
}

File: app/src/main/java/com/jesusc24/xroadsthroughthecastle/data/dao/ChatDAO.kt
package com.jesusc24.xroadsthroughthecastle.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.jesusc24.xroadsthroughthecastle.data.model.Chat

@Dao
interface ChatDAO {
    @Insert
    fun insert(chat: Chat?): Long

    @Query("delete from chat where id = :id")
    fun delete(id: String?)

    @Query("select * from chat order by nombre asc")
    fun select(): List<Chat?>?
}